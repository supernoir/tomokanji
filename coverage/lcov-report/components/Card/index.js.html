<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Card/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Card</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
&nbsp;
export default class Card extends Component {
	constructor(props) {
		super(props);
		this.state = {
			counter    : 0,
			cardData   : this.props.data,
			currentId  : 0,
			currentData: this.props.data
		};
		this.handleClick.bind(this);
	}
&nbsp;
<span class="fstat-no" title="function not covered" >	ha</span>ndleClick(e) {
<span class="cstat-no" title="statement not covered" >		this.setState({ counter: 2 });</span>
	}
&nbsp;
	render() {
		return (
			&lt;div key={this.props.id} className="card"&gt;
				&lt;div className="card-head"&gt;
					&lt;div className="card-kunyomi"&gt;
						&lt;p className="card-reading-type"&gt;訓読み&lt;/p&gt;
						&lt;p className="card-reading-content"&gt;{this.props.kunyomihiragana.length &gt; 0 ? this.props.kunyomihiragana : <span class="branch-1 cbranch-no" title="branch not covered" >'訓読みはありませんでした'}</span>&lt;/p&gt;
					&lt;/div&gt;
					&lt;div className="card-onyomi"&gt;
						&lt;p className="card-reading-type"&gt;音読み&lt;/p&gt;
						&lt;p className="card-reading-content"&gt;{this.props.onyomihiragana}&lt;/p&gt;
					&lt;/div&gt;
				&lt;/div&gt;
				&lt;div className="card-body"&gt;
					&lt;div className="card-kanji"&gt;
						&lt;h1 className="card-kanji-content"&gt;{this.props.kanji}&lt;/h1&gt;
					&lt;/div&gt;
				&lt;/div&gt;
				&lt;div className="card-footer"&gt;
					&lt;div className="card-meaning"&gt;{this.props.meaning}&lt;/div&gt;
					&lt;div className="card-details"&gt;{this.props.detail}&lt;/div&gt;
					&lt;div className="card-classandcount"&gt;
						&lt;p className="card-kanjicount"&gt;
							{`${this.props.current + 1}/${this.props.total + 1}`}
						&lt;/p&gt;
						&lt;p className="card-classification"&gt;
							{this.props.yomi}
						&lt;/p&gt;
					&lt;/div&gt;
				&lt;/div&gt;
			&lt;/div&gt;
		);
	}
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Sep 13 2019 08:40:22 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
