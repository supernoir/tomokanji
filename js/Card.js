import React, { Component } from 'react'

export default class Card extends Component {
  constructor(props) {
    super(props)
    this.state = {
      counter: 0,
      cardData: this.props.data,
      currentId: 0,
      currentData: this.props.data
    }
    this.handleClick.bind(this)
  }

  handleClick(e) {
    this.setState({ counter: 2 })
  }

  render() {
    return (
      <div key={this.props.id} className="card">
        <div className="card-head">
          <div className="card-kunyomi">
            <p className="card-reading-type">訓読み</p>
            <p className="card-reading-content">
              {this.props.kunyomihiragana}
            </p>
            <p className="card-reading-romaji">{this.props.kunyomiromaji}</p>
          </div>
          <div className="card-onyomi">
            <p className="card-reading-type">音読み</p>
            <p className="card-reading-content">
              {this.props.onyomihiragana}
            </p>
            <p className="card-reading-romaji">
              {this.props.onyomiromaji}
            </p>
          </div>
        </div>
        <div className="card-body">
          <div className="card-kanji">
            <h1 className="card-kanji-content">{this.props.kanji}</h1>
          </div>
        </div>
        <div className="card-footer">
          <div className="card-meaning">{this.props.meaning}</div>
          <div className="card-details">{this.props.detail}</div>
        </div>
      </div>
    )
  }
}
