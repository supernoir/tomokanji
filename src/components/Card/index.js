import React, { Component } from 'react';
import {
	StyledCard,
	StyledCardClassification,
	StyledCardClassAndCount,
	StyledCardDetails,
	StyledCardMeaning,
	StyledCardFooter,
	StyledCardKanjiContent,
	StyledCardKanji,
	StyledCardBody,
	StyledCardKunyomiReadingContent,
	StyledCardOnyomiReadingContent,
	StyledCardKunyomiReadingType,
	StyledCardOnyomiReadingType,
	StyledCardOnyomi,
	StyledCardKunyomi,
	StyledCardHead
} from './style';

export default class Card extends Component {
	constructor(props) {
		super(props);
		this.state = {
			cardData   : this.props.data,
			currentId  : 0,
			currentData: this.props.data
		};
	}

	render() {
		return (
			<StyledCard key={this.props.id}>
				<StyledCardHead>
					<StyledCardKunyomi>
						<StyledCardKunyomiReadingType>
							訓読み
						</StyledCardKunyomiReadingType>
						<StyledCardKunyomiReadingContent>
							{this.props.kunyomihiragana.length > 0 ? this.props.kunyomihiragana : '訓読みはありませんでした'}
						</StyledCardKunyomiReadingContent>
					</StyledCardKunyomi>

					<StyledCardOnyomi>
						<StyledCardOnyomiReadingType>
							音読み
						</StyledCardOnyomiReadingType>
						<StyledCardOnyomiReadingContent>
							{this.props.onyomihiragana.length > 0 ? this.props.onyomihiragana : '音読みはありませんでした'}
						</StyledCardOnyomiReadingContent>
					</StyledCardOnyomi>
				</StyledCardHead>

				<StyledCardBody>
					<StyledCardKanji>
						<StyledCardKanjiContent>{this.props.kanji}</StyledCardKanjiContent>
					</StyledCardKanji>
				</StyledCardBody>

				<StyledCardFooter>
					<StyledCardMeaning>{this.props.meaning}</StyledCardMeaning>
					<StyledCardDetails>{this.props.detail}</StyledCardDetails>
					<StyledCardClassAndCount>
						{`${this.props.current + 1}/${this.props.total + 1}`}
						<StyledCardClassification>
							{this.props.yomi}
						</StyledCardClassification>
					</StyledCardClassAndCount>
				</StyledCardFooter>

			</StyledCard>
		);
	}
}
