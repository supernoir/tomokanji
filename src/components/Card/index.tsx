import React, { Component, ReactText, ReactChild } from 'react';
import {
	StyledCard,
	StyledCardClassification,
	StyledCardClassAndCount,
	StyledCardDetails,
	StyledCardMeaning,
	StyledCardFooter,
	StyledCardKanjiContent,
	StyledCardKanji,
	StyledCardBody,
	StyledCardKunyomiReadingContent,
	StyledCardOnyomiReadingContent,
	StyledCardKunyomiReadingType,
	StyledCardOnyomiReadingType,
	StyledCardOnyomi,
	StyledCardKunyomi,
	StyledCardHead
} from './style';

interface YomiMissingProps {
	yomi: string;
}

interface CardProps {
	id: number;
	kunyomi?: string;
	onyomi?: string;
	kanji: string;
	english: string;
	example: string;
	yomi: string | ReactText;
	current: number;
	total: number;

}

const YomiMissing: React.FC<YomiMissingProps> = ({ yomi }) => <div>{`(シ_ _)シ ${yomi}はありませんでした`}</div>

export default class Card extends Component<CardProps, {}> {
	constructor(props: CardProps) {
		super(props);
		this.state = {
			currentId: 0,
		};
	}

	render() {
		return (
			<StyledCard key={this.props.id}>
				<StyledCardHead>
					<StyledCardKunyomi>
						<StyledCardKunyomiReadingType>
							訓読み
						</StyledCardKunyomiReadingType>
						<StyledCardKunyomiReadingContent>
							{this.props.kunyomi ? <p>{this.props.kunyomi}</p> : <YomiMissing yomi={'訓読み'} />}
						</StyledCardKunyomiReadingContent>
					</StyledCardKunyomi>

					<StyledCardOnyomi>
						<StyledCardOnyomiReadingType>
							音読み
						</StyledCardOnyomiReadingType>
						<StyledCardOnyomiReadingContent>
							{this.props.onyomi ? <p>{this.props.onyomi}</p> : <YomiMissing yomi={'音読み'} />}
						</StyledCardOnyomiReadingContent>
					</StyledCardOnyomi>
				</StyledCardHead>

				<StyledCardBody>
					<StyledCardKanji>
						<StyledCardKanjiContent>{this.props.kanji}</StyledCardKanjiContent>
					</StyledCardKanji>
				</StyledCardBody>

				<StyledCardFooter>
					<StyledCardMeaning>{this.props.english}</StyledCardMeaning>
					<StyledCardDetails>{this.props.example}</StyledCardDetails>
					<StyledCardClassAndCount>
						{`${this.props.current + 1}/${this.props.total + 1}`}
						<StyledCardClassification>
							{this.props.yomi}
						</StyledCardClassification>
					</StyledCardClassAndCount>
				</StyledCardFooter>

			</StyledCard>
		);
	}
}
