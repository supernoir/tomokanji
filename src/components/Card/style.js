import styled from 'styled-components';

export const StyledCard = styled.div`
  background: ${props => props.theme.black};
  background: rgba(0,0,0, 0.4);
  color: ${props => props.theme.black};
  padding: 1em;
  border-radius: .4em;
`;

export const StyledCardClassification = styled.p`
  margin-left: 0.6em;
  border-radius: 0.4em;
  padding: 0.2em 0.4em;
  background: ${props => props.theme.white};
  color: ${props => props.theme.black};
`;

export const StyledCardKanjiCount = styled.p`
  background: pink;
`;

export const StyledCardClassAndCount = styled.div`
  border-radius: 4px;
  border: 1px solid ${props => props.theme.white};
  margin-top: 0.8em;
  min-width: 4em;
  padding: 0.2em 0.8em;
  color: ${props => props.theme.white};
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const StyledCardDetails = styled.div`
  width: 100%;
  font-size: 1.2em;
  font-weight: 300em;
  color: ${props => props.theme.tertiary};
`;

export const StyledCardMeaning = styled.div`
  width: 100%;
  font-size: 1.8em;
  font-weight: 500;
`;

export const StyledCardFooter = styled.div`
  background: rgba(0,0,0,0.2);
  color: ${props => props.theme.white};
  padding: 2em 0.4em;
  border-radius: .4em;
  display: flex;
  flex-flow: column;
  align-items: center;
  `;

export const StyledCardKanjiContent = styled.h1`
  color: ${props => props.theme.black};
  font-size: 12em;
  line-height: 0em;
  font-weight: 300;
`;

export const StyledCardKanji = styled.div`
  background: ${props => props.theme.white};
  border-radius: .4em;
  text-align: center;
  margin: 0;
  padding: 2em;
`;

export const StyledCardBody = styled.div`
  margin: 1em 0;
  padding: 0;
`;

export const StyledCardOnyomiReadingContent = styled.p`
  font-size: 2em;
  line-height: 1.2em;
  text-shadow: 0 1px 0 ${props => props.theme.onyomi};
  background: rgba(220, 120, 80, 0.4);
  border-radius: 1em;
`;
export const StyledCardKunyomiReadingContent = styled.p`
  font-size: 2em;
  line-height: 1.2em;
  text-shadow: 0 1px 0 ${props => props.theme.kunyomi};
  background: rgba(80, 180, 160, 0.4);
  border-radius: 1em;
`;

export const StyledCardOnyomiReadingType = styled.p`
  color: ${props => props.theme.white};
  font-size: 0.8em;
  line-height: 0em;
  text-shadow: 0 1px 2px ${props => props.theme.darkonyomi};
`;

export const StyledCardKunyomiReadingType = styled.p`
  color: ${props => props.theme.white};
  font-size: 0.8em;
  line-height: 0em;
  text-shadow: 0 1px 2px ${props => props.theme.darkkunyomi};
`;

export const StyledCardOnyomi = styled.div`
  background: ${props => `linear-gradient(-30deg, ${props.theme.onyomi}, ${props.theme.lightonyomi})`};
  color: ${props => props.theme.white};
  padding: 1em;
  margin-left: 0.5em;
  width: 50%;
  height: 100%;
  border-radius: 0.2em;
  border-top: 6px solid ${props => props.theme.onyomi};
`;

export const StyledCardKunyomi = styled.div`
  background: ${props => `linear-gradient(-30deg, ${props.theme.kunyomi}, ${props.theme.lightkunyomi})`};
  color: ${props => props.theme.white};
  padding: 1em;
  margin-left: 0.5em;
  width: 50%;
  height: 100%;
  border-radius: 0.2em;
  border-top: 6px solid ${props => props.theme.kunyomi};
`;

export const StyledCardHead = styled.div`
  display: flex;
  align-content: space-between;
  margin: 0;
  height: 100%;
`;
