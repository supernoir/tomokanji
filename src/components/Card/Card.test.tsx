import React from 'react';
import Card from './index';
import Renderer from 'react-test-renderer'

const mockData = [
	{
		id: 0,
		kanji: '一',
		onyomi: 'いち、いつ',
		kunyomi: 'ひと、 ひと（つ）',
		english: 'one',
		yomi: '音',
		example: '一どでいいから、宇宙旅行をしてみたい。'
	},
	{
		id: 1,
		kanji: '二',
		onyomi: 'に',
		kunyomi: 'ふた、ふた（つ）',
		english: 'two',
		yomi: '音',
		example: '勉強を二のつぎにしてあそびに行く。'
	}
];

describe('Card', () => {
	test('should match the snapshot', () => {
		const component = Renderer.create(
			<Card
				id={mockData[0].id}
				kanji={mockData[0].kanji}
				kunyomi={mockData[0].kunyomi}
				onyomi={mockData[0].onyomi}
				english={mockData[0].english}
				example={mockData[0].example}
				yomi={mockData[0].yomi}
				current={1}
				total={200}
			/>
		);
		expect(component).toMatchSnapshot();
	});
});
