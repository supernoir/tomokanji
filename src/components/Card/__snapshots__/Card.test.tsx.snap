// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<div
  className="sc-bdVaJa gFIyPa"
>
  <div
    className="sc-jzJRlG bXsGVE"
  >
    <div
      className="sc-fjdhpX gwonBs"
    >
      <p
        className="sc-VigVT dakiqV"
      >
        訓読み
      </p>
      <p
        className="sc-gZMcBi dhGGRZ"
      >
        <p>
          ひと、 ひと（つ）
        </p>
      </p>
    </div>
    <div
      className="sc-jTzLTM bTMtpE"
    >
      <p
        className="sc-gqjmRU dfnrGQ"
      >
        音読み
      </p>
      <p
        className="sc-iwsKbI bdyKZ"
      >
        <p>
          いち、いつ
        </p>
      </p>
    </div>
  </div>
  <div
    className="sc-dnqmqq etplQW"
  >
    <div
      className="sc-htoDjs gEHvYl"
    >
      <h1
        className="sc-gzVnrw iozBrE"
      >
        一
      </h1>
    </div>
  </div>
  <div
    className="sc-bZQynM eerjQp"
  >
    <div
      className="sc-EHOje eOWCyW"
    >
      one
    </div>
    <div
      className="sc-ifAKCX bYrwdv"
    >
      一どでいいから、宇宙旅行をしてみたい。
    </div>
    <div
      className="sc-bxivhb grvtRF"
    >
      2/201
      <p
        className="sc-bwzfXH cGawet"
      >
        音
      </p>
    </div>
  </div>
</div>
`;
